### ISSUE TITLE
Can't authenticate with MFA

### ISSUE THREAD

**Issue by alexconlin** [awslabs/aws-shell#168]
i am using an authentication profile that requires an MFA token. When i open aws-shell and type `kinesis get-shard-iterator --stream-name ` it prints `Enter MFA code:` in the command line but when I enter it nothing happens. If I then try to use the aws-shell it displays an exception and then runs really slowly and I have to quit it.

**Comment by donnemartin** [comment]
Hi @alexconlin, can you post the exception you get in the `aws-shell`?

Can you also try the same steps using the `awscli` and check if the problem exists there?

**Comment by donnemartin** [comment]
Thanks for the additional info Alex.  This should be helpful when we get a chance to dig in further.

**Comment by rickard-von-essen-iz** [comment]
This did work much better for me when running from `master` b9975fa9f0bbb92309879f4417ec87ee23f03846 instead of using the `0.1.1` from Homebrew.

**Comment by gene1wood** [comment]
I've also encountered this with `boto3` though I'm unclear on what's causing it. In adding debug lines to botocore what I'm seeing is if I print the contents of the `token_code` [immediately after the prompter gathers it from the user](https://github.com/boto/botocore/blob/bdc971d5a9d0a1e6574bf5203450511ab228e8d5/botocore/credentials.py#L979) it is not the number typed in but instead a string containing high ascii characters. I've confirmed when this happens the prompter is `getpass.unix_getpass`. If I then go and try to reproduce this in the python REPL in the same virtualenv it works fine. I get this on an old version of botocore (1.5.16) and I think this works in current versions of botocore. It might relate to the relationship between the current boto3 and old botocore, as using this old botocore (with old boto3) previously did not cause this.

Anyhow, I'm just going to upgrade botocore and I think that will resolve this for me.
